src/java/org/apache/cassandra/repair/messages/RepairMessage.java

```diff
@@ -232,4 +298,15 @@ private static ErrorHandling errorHandlingSupported(SharedContext ctx, InetAddre
             return ErrorHandling.TIMEOUT;
         return ErrorHandling.NONE;
     }
+
+    public static void sendFailureResponse(SharedContext ctx, Message<?> respondTo)
```

Comment:
thats fine, we drop in the messaging layer... these are *replies* which require we have Callbacks define for the message number, so the below block is what we hit in 4.x

```
RequestCallbacks.CallbackInfo callbackInfo = MessagingService.instance().callbacks.remove(message.id(), message.from());
        if (callbackInfo == null)
        {
            String msg = "Callback already removed for {} (from {})";
            logger.trace(msg, message.id(), message.from());
            Tracing.trace(msg, message.id(), message.from());
            return;
        }
```

Reply:
> but why reply if it's just going to be dropped?

in order to know this we need to check gossip, which maybe stale.  So If we think its 4.x but its actually 5.x then we avoid sending the ack/nack and it will retry... 

Now, since we don't 100% support mixed version repairs, by not caring we make this simpler to reason about with the cost of empty dropped messages while in mixed mode (again, we don't support)... so this patch does not break mixed mode, but isn't 100% ideal in mixed mode...